prow/config/config.go

```diff
@@ -348,6 +349,13 @@ type Branding struct {
 	HeaderColor string `json:"header_color,omitempty"`
 }
 
+// Build is config for the knative build controller.
+type Build struct {
```

Comment:
@fejta do you want this as a top-level field or for the controller?

Reply:
I embedded Controller into Build and defined also a temporary flag which is off.

Reply:
At this commit we now have three values in `config.yaml`:
`plank.AllowCancellations`
`build.AllowCancellations`
`build.AllowBuildCancellations`

Embedding Controller into Build ensures that these two fields have the same possible values, but still requires configuring them independently. 

I am arguing that we should end up with a single `AllowCancellations` field in `config.yaml` that both build and plank read.

a) If there is an independent `config.build.AllowCancellations` field, we do not also need an `AllowBuildCancellations` field.
b) If build reads `config.plank.AllowCancellations`, I would recommend moving the feature flag into an actual `--flag` in `prow/cmd/build/main.go` (not a `config.yaml` field).
  - This way in the future when we delete this flag because we now assume the feature is on, when someone forgets to drop this flag it will just blow up the build controller (bad flags are local to the build binary), rather than every component (bad config prevents every component from loading)

We should only have an independent `config.build.AllowCancellations` field if we think there is some reason/need to configure this value differently from plank (I do not see this need).

Reply:
I introduced a flag in `build` which gates the usage of `config.plank.AllowCancellations`, and removed completely the build config.